Testing/7 player expansion

ODE-BOY opened the pull request:
# Description
This PR fixes #[284]

Added Domain and Integration tests for 7 player lobbys.
Added 7 tests. 1 Domain, 6 Integration.

# Demo
![image](https://user-images.githubusercontent.com/48495086/113683607-80274a80-9718-11eb-8d9b-82c4650f930a.png)



# Type of change:
* New feature (Non-breaking change that adds functionality)

ODE-BOY committed:
Update the Unit Tests

Refactored The Domain Model test to test a six and seven person lobby.

ODE-BOY committed:
Update Integration Tests

Expanded the 6 player tests to test both for 6 and 7 player games for Game Start Events, Lobby Events, Vote Events and Night Time Vote Events.

ODE-BOY committed:
Update StateChangeEvents.integration.test.js

Added 7 player versions of the integration tests that currently test 6 players.

ODE-BOY committed:
Update the StartGame  Functionality

Added Start Game Functionality for 6 and 7 person lobbys to all the integration tests

ODE-BOY committed:
Merge branch 'a2-main' into testing/7-player-expansion

RawiriHohepa commented on backend/test/integration/GameStartEvents.integration.test.js:
```diff
@@ -34,7 +39,17 @@ describe('GameStartEvents integration tests', () => {
         }
         /* eslint-enable no-await-in-loop */
-        await startGame(clientSockets);
+        await startGameSixPlayer(clientSockets);
```
Can you combine the two startGame functions and pass through the number as a parameter?

ODE-BOY replied:
Good Idea. Ill make those changes.

RawiriHohepa commented on backend/test/integration/StateChangeEvents.integration.test.js:
```diff
@@ -132,7 +175,44 @@ describe('DayStateEvents integration tests', () => {
+        done();
+    });
+
+    test('integration test start-day 7 players', async (done) => {
```
Is there a way to reduce the duplicate code here and somehow pass in the number of players?

ODE-BOY replied:
Done and Done.

kyzh0 reviewed (dismissed):
All tests pass, LGTM!

RawiriHohepa reviewed (changes requested):
LGTM, just wondering if there's a way to reduce the amount of duplication, especially if we plan to make more tests with different numbers of players

ysway commented on backend/test/integration/LobbyEvents.integration.test.js:
```diff
@@ -25,17 +25,25 @@ describe('lobby events tests', () => {
     /**
-     * This is a integration test for all lobby events.
-     * It encompases creating a game, Max players joining and reseting the game
+     * These are a integration tests for all lobby events.
```
These are ~~a~~ integration tests for all lobby events.

ODE-BOY replied:
Good catch ill fit it up

ysway reviewed (commented):
grammar

ODE-BOY committed:
Revert "Update the StartGame  Functionality"

This reverts commit 11e0faf92ab5cebc1505c43ff74e0b0262105d2f.

ODE-BOY committed:
Alteres the start game functionality to reduce code repetition

Start game function now takes the number of players as an input to be used for integration tests

ODE-BOY committed:
Removed Repeated code in Integration tests

Created loops in Start, Lobby,  State Change, Vote and Night Time Vote Event Tests to test all the games potential lobby sizes

RawiriHohepa commented on backend/test/integration/GameStartEvents.integration.test.js:
```diff
@@ -28,13 +28,16 @@ describe('GameStartEvents integration tests', () => {
-        done();
-    });
+    // Run integration tests for all possible lobby sizes.
+    for (let playerCount = 6; playerCount <= 7; playerCount += 1) {
```
This is great, especially making a separate test() for each playerCount. We just need to remember to update these values when the maximum player count changes

RawiriHohepa reviewed (approved):
Looks great, much better without the code duplication

kyzh0 reviewed (approved):
LGTM!

kyzh0 committed:
Merge branch 'a2-main' into testing/7-player-expansion
